plaso/parsers/viminfo.py

```diff
@@ -0,0 +1,529 @@
+      event_data.history_type = 'Input Line History'
+      event_data.item_number = index
+
+      timestamp = item[5]
+      date_time = dfdatetime_posix_time.PosixTime(timestamp=timestamp)
+      event = time_events.DateTimeValuesEvent(
+          date_time, definitions.TIME_DESCRIPTION_RECORDED)
+      parser_mediator.ProduceEventWithEventData(event, event_data)
+
+  def _ParseDebugLineHistory(self, parser_mediator, structure):
+    """Parses debug line history items and creates VimInfoEventData objects.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          lines of text that match a debug line history
+    """
+    for index, item in enumerate(structure.get('debug_line_history_items', [])):
+      event_data = VimInfoEventData()
+      event_data.history_value = item[1]
+      event_data.history_type = 'Debug Line History'
+      event_data.item_number = index
+
+      timestamp = item[5]
+      date_time = dfdatetime_posix_time.PosixTime(timestamp=timestamp)
+      event = time_events.DateTimeValuesEvent(
+          date_time, definitions.TIME_DESCRIPTION_RECORDED)
+      parser_mediator.ProduceEventWithEventData(event, event_data)
+
+  def _ParseRegistersHistory(self, parser_mediator, structure):
+    """Parses register history items and creates VimInfoEventData objects.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          lines of text that match a Register history
+    """
+    for item in structure.get('registers_items', []):
+      event_data = VimInfoEventData()
+      event_data.history_value = '\n'.join(item[4])
+      event_data.history_type = 'Register'
+      event_data.item_number = item[1]
+
+      timestamp = item[12]
+      date_time = dfdatetime_posix_time.PosixTime(timestamp=timestamp)
+      event = time_events.DateTimeValuesEvent(
+          date_time, definitions.TIME_DESCRIPTION_RECORDED)
+      parser_mediator.ProduceEventWithEventData(event, event_data)
+
+  def _ParseFilemarksHistory(self, parser_mediator, structure):
+    """Parses filemarks history items and creates VimInfoEventData objects.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          lines of text that match a Filemarks history
+    """
+    for index, item in enumerate(structure.get('filemarks_items', [])):
+      event_data = VimInfoEventData()
+      event_data.filename = item[4].strip()
+      event_data.history_type = 'File mark'
+      event_data.item_number = index
+
+      timestamp = item[5][5]
```

Comment:
@sydp is this correct?

Reply:
Yes, index 5 is the parsed result of the second line of the file mark and then index of that is the time stamp.